File: 2059-unique-length-3-palindromic-subsequences/unique-length-3-palindromic-subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        vector<int>first(26,-1);
        vector<int>last(26,-1);
        int count=0;
        for(int i=0;i<s.size();i++){
            if(first[s[i]-'a']==-1)
            {
                first[s[i]-'a']=i;
            }
            last[s[i]-'a']=i;
        }
        for(char ch='a';ch<='z';ch++){
            int left=first[ch-'a'];
            int right=last[ch-'a'];
            set<char>st;

            for(int i=left+1;i<right;i++){
                st.insert(s[i]-'a');


            }
            count+=st.size();
        }
        return count;
        
    }
};

File: 2059-unique-length-3-palindromic-subsequences/unique-length-3-palindromic-subsequences.java
class Solution {
    public int countPalindromicSubsequence(String s) {
        int n=s.length();
        int first[]=new int[26];
        int last[]=new int[26];
        
        int freq[]=new int[n+1];
        Arrays.fill(first,-1);
        Arrays.fill(last,-1);
        for(int i=0;i<n;i++){
            char ch=s.charAt(i);
            if(first[ch-'a']==-1){
                first[ch-'a']=i;
            }
            last[ch-'a']=i;
        }
        int count=0;
        for(char ch='a';ch<='z';ch++){
            if(first[ch-'a']!=-1 && last[ch-'a']!=-1){
                HashSet<Character>st=new HashSet<>();
                for(int i=first[ch-'a']+1;i<last[ch-'a'];i++){
                    st.add(s.charAt(i));
                }
                count+=st.size();

            }
        }
        return count;
    

        
    }
}